extractor/extractor.cpp

```diff
@@ -262,5 +322,309 @@ int extractor::run()
         SimpleLogger().Write(logWARNING) << e.what();
         return 1;
     }
+
     return 0;
 }
+
+unsigned extractor::CalculateEdgeChecksum(std::unique_ptr<std::vector<EdgeBasedNode>> node_based_edge_list)
+{
+    RangebasedCRC32 crc32;
+    if (crc32.using_hardware())
+    {
+        SimpleLogger().Write() << "using hardware based CRC32 computation";
+    }
+    else
+    {
+        SimpleLogger().Write() << "using software based CRC32 computation";
+    }
+
+    const unsigned crc32_value = crc32(*node_based_edge_list);
+    SimpleLogger().Write() << "CRC32: " << crc32_value;
+
+    return crc32_value;
+}
+
+/**
+    \brief Setups scripting environment (lua-scripting)
+    Also initializes speed profile.
+*/
+void extractor::SetupScriptingEnvironment(lua_State *lua_state, SpeedProfileProperties &speed_profile)
```

Comment:
Hmm I know this code only got moved around, but I wonder why we take `lua_state` by pointer, when a reference would be good enough. Either that, or at least `assert(lua_state);`.